tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -491,27 +492,24 @@ void RNTupleMerger::MergeCommonColumns(RClusterPool &clusterPool, DescriptorId_t
    const auto &clusterDesc = mergeData.fSrcDescriptor->GetClusterDescriptor(clusterId);
 
    for (const auto &column : commonColumns) {
-      const auto &columnId = column.fInputId;
-      R__ASSERT(clusterDesc.ContainsColumn(columnId));
+      R__ASSERT(clusterDesc.ContainsColumn(column.fInputId));
 
-      const auto &columnDesc = mergeData.fSrcDescriptor->GetColumnDescriptor(columnId);
+      const auto &columnDesc = mergeData.fSrcDescriptor->GetColumnDescriptor(column.fInputId);
       const auto srcColElement = column.fInMemoryType
                                     ? GenerateColumnElement(*column.fInMemoryType, columnDesc.GetType())
                                     : RColumnElementBase::Generate(columnDesc.GetType());
       const auto dstColElement = column.fInMemoryType ? GenerateColumnElement(*column.fInMemoryType, column.fColumnType)
                                                       : RColumnElementBase::Generate(column.fColumnType);
 
       // Now get the pages for this column in this cluster
-      const auto &pages = clusterDesc.GetPageRange(columnId);
+      const auto &pages = clusterDesc.GetPageRange(column.fInputId);
 
       RPageStorage::SealedPageSequence_t sealedPages;
       sealedPages.resize(pages.fPageInfos.size());
 
       // Each column range potentially has a distinct compression settings
-      const auto colRangeCompressionSettings = clusterDesc.GetColumnRange(columnId).fCompressionSettings;
-      const bool needsCompressionChange = mergeData.fMergeOpts.fCompressionSettings != kUnknownCompressionSettings &&
-                                          colRangeCompressionSettings != mergeData.fMergeOpts.fCompressionSettings;
-
+      const auto colRangeCompressionSettings = clusterDesc.GetColumnRange(column.fInputId).fCompressionSettings;
+      const bool needsCompressionChange = colRangeCompressionSettings != mergeData.fMergeOpts.fCompressionSettings;
```

Comment:
For `needsCompressionChange`, you drop the condition `mergeData.fMergeOpts.fCompressionSettings != kUnknownCompressionSettings` - is this intentional? Again, not related to the column id changes